src/Materials/shaderMaterial.ts

```diff
@@ -1432,7 +1448,7 @@ export class ShaderMaterial extends Material {
      * @returns a new material
      */
     public static Parse(source: any, scene: Scene, rootUrl: string): ShaderMaterial {
-        var material = SerializationHelper.Parse(() => new ShaderMaterial(source.name, scene, source.shaderPath, source.options), source, scene, rootUrl);
+        var material = SerializationHelper.Parse(() => new ShaderMaterial(source.name, scene, source.shaderPath, source.options, source.storeEffectOnSubMeshes), source, scene, rootUrl);

```

Comment:
Nit: const

Reply:
also would forcing storeEffectOnSubMesh to false works in the default case ?

Reply:
I'm not sure I understand.

`source.storeEffectOnSubMeshes` can be `undefined` if parsing a material that has been serialized before this PR, in which case the `storeEffectOnSubMeshes` parameter will take the default value defined in the constructor, which is `true`. That is the value we should set by default I think, else people writing code for WebGPU and using the non compat mode will experience lower perf when using `ShaderMaterial`. I think a lot of people won't be aware of it (even if we document it and explain what to do to fix it), so in my opinion it's better to set the value so that the perf are the best out of the box.

Reply:
Yup totally agree I was mostly wondering in case they were using in a way which would require false which would then be none back compat if it could exist ?

Reply:
Yes, it can be a breaking change in some cases: I have added some explanations in the "Breaking Changes" section of the what's new file. I expect that this will only affect a few people, though.